test/characterization/generate_data.py

```diff
@@ -0,0 +1,325 @@
+    - 't2'
+    - 'omega'
+
+    Args:
+       filename - name of the json file.
+    """
+
+    backend_result, xdata, qubits, t2_value, omega = t2star_circuit_execution()
+
+    data = {
+        'backend_result': backend_result.to_dict(),
+        'xdata': xdata.tolist(),
+        'qubits': qubits,
+        't2': t2_value,
+        'omega': omega
+        }
+
+    with open(filename, 'w') as handle:
+        json.dump(data, handle)
+
+
+def zz_circuit_execution() -> Tuple[qiskit.result.Result,
+                                    np.array,
+                                    List[int],
+                                    List[int],
+                                    float,
+                                    float]:
+    """
+    Create ZZ circuits and simulate them.
+
+    Returns:
+        *   Backend result.
+        *   xdata.
+        *   Qubits for the ZZ measurement.
+        *   Spectators.
+        *   ZZ parameter that used in the circuit creation
+        *   Frequency.
+    """
+
+    num_of_gates = np.arange(0, 60, 10)
+    gate_time = 0.1
+    qubits = [0]
+    spectators = [1]
+
+    # Generate experiments
+    circs, xdata, omega = zz_circuits(num_of_gates,
+                                      gate_time, qubits,
+                                      spectators, nosc=2)
+
+    # Set the simulator with ZZ
+    zz_value = 0.1
+    zz_unitary = np.eye(4, dtype=complex)
+    zz_unitary[3, 3] = np.exp(1j*2*np.pi*zz_value*gate_time)
+    error = coherent_unitary_error(zz_unitary)
+    noise_model = NoiseModel()
+    noise_model.add_nonlocal_quantum_error(error, 'id', [0], [0, 1])
+
+    # Run the simulator
+    backend = qiskit.Aer.get_backend('qasm_simulator')
+    shots = 100
+
+    backend_result = qiskit.execute(circs, backend,
+                                    shots=shots,
+                                    seed_simulator=SEED,
+                                    noise_model=noise_model,
+                                    optimization_level=0).result()
+
+    return backend_result, xdata, qubits, spectators, zz_value, omega
+
+
+def generate_data_zz(filename):
+    """
+    Create ZZ circuits and simulate them, then write the results in a json file.
+    The file will contain a dictionary with the following keys:
+    - 'backend_result', value is stored in the form of a dictionary.
+    - 'xdata', value is stored as a list (and not as a numpy array).
+    - 'qubits', these are the qubits for the ZZ measurement.
+    - 'spectators'
+    - 'zz'
+    - 'omega'
+
+    Args:
+       filename - name of the json file.
+    """
+
+    backend_result, xdata, qubits, spectators, zz_value, omega = zz_circuit_execution()
+
+    data = {
+        'backend_result': backend_result.to_dict(),
+        'xdata': xdata.tolist(),
+        'qubits': qubits,
+        'spectators': spectators,
+        'zz': zz_value,
+        'omega': omega
+        }
+
+    with open(filename, 'w') as handle:
+        json.dump(data, handle)
```

Comment:
We should add an executor to this to run all the generation methods (or a subset by just calling this. For example something like:

```suggestion
        json.dump(data, handle)
        
if __name__ == '__main__':
    dir_name = os.path.dirname(os.path.abspath(__file__))
    generate_data_zz(os.path.join(dir_name, 'zz_data.json'))
    generate_data_t2star(os.path.join(dir_name, 't2star_data.json'))
    generate_data_t2(os.path.join(dir_name, 't2_data.json'))
    generate_data_t1(os.path.join(dir_name, 't1_data.json'))
```

The only other thing would be to add argparse and add a parameter with an arbitrary number of nargs and use that for each dataset name. So if it's set it loops over them and only updates those files.